extensions/amp-analytics/0.1/amp-analytics.js

```diff
@@ -694,4 +742,95 @@ export class AmpAnalytics extends AMP.BaseElement {
   }
 }
 
+/**
+ * Keeps track of the third-party vendors' cross-domain iframes. Only one
+ * static instance will be created, which will keep and manage ALL amp-analytics
+ * tags.
+ */
+class AmpAnalyticsFrames {
+  constructor() {
+    this.requestsFrameMap = map();
+  }
+
+  /**
+   * Create a cross-domain iframe
+   * @param {!HTMLDocument} parent  The document node of the parent page
+   * @param {!string} requestsFrameUrl  The URL, including data hash if
+   * applicable, of the cross-domain iframe
+   * @return {!Element}
+   */
+  create(parent, requestsFrameUrl) {
+    // Warning: the scriptSrc URL below is only temporary. Don't check
+    // in before getting resolution on that.
+    const frame = createElementWithAttributes(parent, 'iframe', {
+      sandbox: 'allow-scripts',
+      name: JSON.stringify({
+        'scriptSrc': '/examples/analytics-3p-remote-frame-helper.js',
+      }),
+    });
+    loadPromise(frame).then(() => {
+      this.setIsReady(requestsFrameUrl);
+    });
+    frame.src = requestsFrameUrl; // Set AFTER loadPromise to avoid race
```

Comment:
If this needs to be done after loadPromise is done, the current implementation will not do that. loadPromise returns a promise, and frame.src will be set immediately after the previous line, whether or not that promise has actually resolved. Is that an issue?